### PR TITLE
Move things to utils, test interchange creation

### PR DESCRIPTION
**Pull request by hannahbaumann**
## Description
This PR seeks to address the following issues:

- [x] #30 
- [x] #31 
- [x] #33 
   - [x] Test for virtual sites: Waiting on Matt's reply, if virtual sites will not be supported any time soon I should add a validator, not allowing tip4p waters
- [x] #24 
- [x] #53

### CODE REVIEW

File: openfe_gromacs/protocols/gromacs_utils/system_creation.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
+import os
+import warnings
+
+import gufe
+from openfe.protocols.openmm_utils import charge_generation, system_creation
```
Comment (IAlibay) [inline openfe_gromacs/protocols/gromacs_utils/system_creation.py:5]: I would suggest not calling two different things system_creation, it's too prone to confusion for developers.

File: openfe_gromacs/tests/protocols/test_create_systems.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+from openfe_gromacs.protocols.gromacs_md.md_methods import GromacsMDProtocol
+from openfe_gromacs.protocols.gromacs_utils import create_systems
+
+
+def test_interchange_gromacs(T4_protein_component, tmpdir):
+    solvent = gufe.SolventComponent()
+    smc_components = {}
+    prot_settings = GromacsMDProtocol.default_settings()
+    # The function expects a settings dict
+    settings = {}
```
Comment (hannahbaumann) [inline openfe_gromacs/tests/protocols/test_create_systems.py:18]: `create_openmm_systems` takes in a settings dict, which in the protocol is created using the `_handle_settings` function. This makes this test not so pretty, as I have to create the dictionary. Should I change the function to instead take in the actual settings objects? Or is this ok?

File: openfe_gromacs/tests/conftest.py [hunk 3]
```diff
@@ -142,6 +143,15 @@ def T4_protein_component():
     return comp
 
 
+@pytest.fixture(scope="session")
+def alanine_dipeptide_component():
+    with resources.files("openfe_gromacs.tests.data") as d:
```
Comment (hannahbaumann) [inline openfe_gromacs/tests/conftest.py:149]: This part doesn't work yet, that's why for now I've been using the test data stored in the openfe repo (e.g. `T4_protein_component` in `resources.files("openfe.tests.data")`).

Comment (IAlibay) [reply openfe_gromacs/tests/conftest.py:149]: I've pushed a commit to fix this, you were missing an `__init__.py` file in the `tests/data` directory.

**Review by IAlibay — CHANGES_REQUESTED** [review]

File: openfe_gromacs/tests/protocols/test_create_systems.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    # check a few atom names to ensure these are not empty sets
+    for atom_name in ("HA", "CH3", "CA", "CB"):
+        assert atom_name in interchange_atom_names
+
+
+def test_user_charges(ethane, tmpdir):
+    solvent = gufe.SolventComponent()
+    off_ethane = ethane.to_openff()
+    off_ethane.assign_partial_charges(partial_charge_method="am1bcc")
```
[the remark this answers is not in the document]
Comment (hannahbaumann) [reply openfe_gromacs/tests/protocols/test_create_systems.py]: Changed this to `am1-mulliken`

File: openfe_gromacs/tests/protocols/test_create_systems.py (hunk 2 above)
Comment (hannahbaumann) [reply openfe_gromacs/tests/protocols/test_create_systems.py]: I changed this to instead pass in the individual settings. Alternatively, I was thinking we could also have an `SystemCreationSettings` class that contains the settings necessary to create the OpenMM system to distinguish with the settings that are related to Gromacs. But I'm not sure if that would be confusing to the user.

File: openfe_gromacs/tests/protocols/test_create_systems.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    )
+    omm_atom_names = [atom.name for atom in omm_topology.atoms()]
+    interchange = create_systems.create_interchange(
+        omm_system, omm_topology, omm_positions, smc_components
+    )
+    interchange_atom_names = [atom.name for atom in interchange.topology.atoms]
+    interchange.to_gro(f"{tmpdir}/test.gro")
+    gro_atom_names = GromacsGroFile(f"{tmpdir}/test.gro").atomNames
+    assert len(omm_atom_names) == len(interchange_atom_names) == len(gro_atom_names)
+    assert omm_atom_names == interchange_atom_names == gro_atom_names
```
[the remark this answers is not in the document]
Comment (hannahbaumann) [reply openfe_gromacs/tests/protocols/test_create_systems.py:41]: I added checks for this, the one for the residue numbers is currently failing as it requires the interchange 0.4 release. Should I pin that release candidate for now?

[the remark this answers is not in the document]
Comment (IAlibay) [reply openfe_gromacs/tests/protocols/test_create_systems.py:41]: Thanks, I don't think there's any rush for us to push this on conda forge or elsewhere, so this is ok. Please do raise an issue to unpin later though!

File: openfe_gromacs/tests/protocols/test_create_systems.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+import gufe
+import numpy as np
+from openmm import NonbondedForce
+from openmm.app import GromacsGroFile, GromacsTopFile
+
+from openfe_gromacs.protocols.gromacs_md.md_methods import GromacsMDProtocol
+from openfe_gromacs.protocols.gromacs_utils import create_systems
+
+
+def test_interchange_gromacs(alanine_dipeptide_component, tmpdir):
```
Comment (IAlibay) [inline openfe_gromacs/tests/protocols/test_create_systems.py:12]: Could you open an issue that effectively says something like adding a single point energy test - interchange should already do this but it looks like folks want us to verify.

File: openfe_gromacs/tests/protocols/test_create_systems.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    with open(f"{tmpdir}/test.gro") as f:
+        gromacs_res_numbers = [int(line[:5].strip()) for line in f.readlines()[2:-1]]
+        assert interchange_res_numbers == gromacs_res_numbers
+
+    # check a few atom names to ensure these are not empty sets
+    for atom_name in ("HA", "CH3", "CA", "CB"):
+        assert atom_name in interchange_atom_names
+
+
+def test_user_charges(CN_molecule, tmpdir):
```
Comment (IAlibay) [inline openfe_gromacs/tests/protocols/test_create_systems.py:62]: Could you add an issue to test other things from our different backends? like NAGL etc... not a priority, just a nice thing we can add later.

**Review by IAlibay — APPROVED** [review]
lgtm thanks! just a couple of issues worth raising for future work.

**Pull request merged**